cli/config/configfile/file.go

```diff
@@ -46,6 +48,65 @@ type ConfigFile struct {
 	Experimental string `json:"experimental,omitempty"`
 }
 
+type configEnvAuth struct {
+	Auth string `json:"auth"`
+}
+
+type configEnv struct {
+	AuthConfigs map[string]configEnvAuth `json:"auths"`
+}
+
+// dockerEnvConfig is an environment variable that contains a JSON encoded
+// credential config. It only supports storing the credentials as a base64
+// encoded string in the format base64("username:pat").
+//
+// Adding additional fields will produce a parsing error.
+//
+// Example:
+//
+//	{
+//		"auths": {
+//			"example.test": {
+//				"auth": base64-encoded-username-pat
+//			}
+//		}
+//	}
+const dockerEnvConfig = "DOCKER_AUTH_CONFIG"
+
+// LoadFromEnv decodes a JSON encoded string into configEnv
+//
+// Unexpected fields will cause a parsing error.
+func (c *configEnv) LoadFromEnv(envConfig string) error {
+	if c.AuthConfigs == nil {
+		c.AuthConfigs = make(map[string]configEnvAuth)
+	}
+	decoder := json.NewDecoder(strings.NewReader(envConfig))
+	decoder.DisallowUnknownFields()
+	if err := decoder.Decode(c); err != nil && !errors.Is(err, io.EOF) {
+		return err
+	}
+	return nil
```

Comment:
We should call `.More()` to verify it's indeed a single JSON document; https://pkg.go.dev/encoding/json#Decoder.More